internal/pkg/manifest/validate.go

```diff
@@ -745,36 +765,54 @@ func (CommandOverride) validate() error {
 	return nil
 }
 
+// validate returns nil if RoutingRuleConfiguration is configured correctly.
+func (r RoutingRuleConfiguration) validate() error {
+	if r.IsEmpty() {
+		return nil
+	}
+	// we consider the fact that primary routing rule is mandatory before you write any additional routing rules.
+	if err := r.MainRoutingRule.validate(); err != nil {
+		return err
+	}
+	// checking this condition here as validate method on ALBRoutingRule only validates common parameters between
+	// MainRoutingRule and AdditionalRoutingRules
```

Comment:
nit: not to represent everyone so plz correct me if others think otherwise😂 because `r. TargetContainerCamelCase` is involved, I feel like it should be fairly obvious to readers why this validation does not happen in `func (r ALBRoutingRule) validate() error {`,
```suggestion
```